### DISCUSSION TITLE
Variable speed pump

### DISCUSSION THREAD

**Discussion by mrdavidscott** [aqualinkd/AqualinkD#280] [Q&A] [answered]
Hello I have a jandy rs8 one touch system. I'm working on upgrading to the aqualinkD and currently have a Hayward TriStar VS SP3200VSP pump. Would aqualinkD be able to control the variable speed of that pump via the RS-485 connection? The Hayward manual shows a RS-485 connection. Or since it's Hayward it's not compatible with jandys system? Currently I'm only able to control the speed via which aux's are on and off.

**Comment by sfeakes** [comment, marked as answer]
Short answer is I don't believe so.

Long answer.
This will depend on a lot of things.  AqualinkD will only control devices through the AqualinkRS control panel, so if the panel does not control the device, aqualinkd can't either. AqualinkD can read anything on the RS485 buss and give you the detailed information, but doesn't actually control the device.
RS485 bus has to have one master and multiple slaves, the master controls everything, and this is the control panel, it can;t be changed.

Depending on the revision, Jandy AqualinkRS panels can control Jandy ePumps and Pentair VSP. Many different VSP manufacturers have a Pentair emulation mode (or an add on board) where they will mimic a Pentair VSP. (Ie Black & Decker VSP have a $20 board you can buy that makes it look like a Pentair VSP).

Jandy AqualinkRS panels from revision `P` have good support for VSP.
So if you can put your Haywood VSP into a Pentair emulation mode, and your Jandy RS panel is rev P of newer, you should get full VSP support.  Anything else and you'll be stuck with the manual relay mode where speed is controlled via what device is turned on.
